### ISSUE TITLE
Firebase integration

### ISSUE THREAD

**Issue by janayama** [segmentio/analytics-react-native#9]
Hey there!

I have one configuration question I did not know where to ask, so feel free to close this if you guys think this is not the place.

Currently integration segment with a react-native app and curious about how segment is integrating this with firebase. I mean, I am adding `@segment/analytics-react-native-firebase` into the project and configuring it with  `.using(firebase)`.

But more than that, do I need to follow the normal integration of firebase with native apps: https://segment.com/docs/destinations/firebase/

Or how is that plugin sending the information to firebase?

Configuring firebase as a destination at the segment page does not ask me for any firebase key for integrating it, so I guess the app talks directly with firebase and not with segment.

Thank you in advance.
Best.

**Comment by YahiaYasLife** [comment]
Hi @fathyb 

Thank you for the details.

I did all the steps but I don't receive any event in my firebase event dashboard.
FYI,  I'm using in my project the [react-native-firebase](https://rnfirebase.io/) dependency.
and I'm using `classpath 'com.google.gms:google-services:4.2.0'`
Also, I'm receiving successfully the events in segment debugger 

Any idea plz ?

**Comment by yamov** [comment]
On iOS just adding the `GoogleService-Info.plist` was not enough for me. The Firebase console was not detecting the connection.

I had to manually call `[FIRApp configure];` in `Appdelegate.m`, as Firebase installation guide instructs:

> To connect Firebase when your app starts up, add the initialisation code below to your main AppDelegate class.

```ObjectiveC
@import UIKit;
@import Firebase;

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application
    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  [FIRApp configure];
  return YES;
}
```

**Comment by twelve17** [comment]
Hello @fathyb , thanks for your high level explanation of how integrations are registered.  I was curious if you could point me to how to configure integrations that have other options?  Specifically, I see that AppBoy's android java config can look like:
```
AppboyConfig appboyConfig = new AppboyConfig.Builder()
                .setApiKey(<key>)
                .setIsFirebaseCloudMessagingRegistrationEnabled(true)
                .setFirebaseCloudMessagingSenderIdKey(<sender id>)
                .build();
        Appboy.configure(this, appboyConfig);
        registerActivityLifecycleCallbacks(new AppboyLifecycleCallbackListener());
```

How would the equivalent be done using the `@segment/analytics-react-native-appboy` integration?  E.g.
`setFirebaseCloudMessagingSenderIdKey`
`setIsFirebaseCloudMessagingRegistrationEnabled`

**Comment by jariwalabhavesh** [comment]
I am total confused and not able to identify what to do. I am not receiving any event on firebase.

What I did :
1. Installed and link @segment/analytics-react-native and i am able to receive event on segment dashboard in source debug
2. I had installed and link @segment/analytics-react-native-firebase and added 

```
import SegmentFirebase from '@segment/analytics-react-native-firebase'

 analytics .setup(Config.SEGMENT_WRITE_KEY, {
        debug: true,
        using: [SegmentFirebase],
        recordScreenViews: false,
        trackAppLifecycleEvents: true,
        trackAttributionData: true,

        android: {
          flushInterval: 60,
          collectDeviceId: true
        },
        ios: {
          trackAdvertising: true,
          trackDeepLinks: true
        }
      })
      .then(() => {
        setCurrentScreen('_home', { routeName: 'home' })
        // firebase.analytics().setUserProperties({ env: Config.ENVIRONMENT })
      }

      )
      .catch(err =>
        console.error('Something went wrong', err)
      )
```

**Comment by ivansnek** [comment]
I know this issue is closed but, @jariwalabhavesh were you able to fix it? I'm experiencing a similar issue, where I'm not receiving Firebase events for Android, iOS looks ok, also Segment events looks oks. Thanks

**Comment by jariwalabhavesh** [comment]
@ivansnek : Yes my issue was fixed. Everything in code was fine. Issue was there in segment app. In segment firebase destination was not enable due to that segment SDK was not sending event

I had just enable it and then event start receiving on firebase as well.
<img width="1170" alt="Screenshot 2020-10-13 at 7 10 39 PM" src="https://user-images.githubusercontent.com/10993666/95868589-02e1d500-0d88-11eb-810e-6c13885b549f.png">

**Comment by JijoBose** [comment]
Could anyone help me to fix this issue for Google Analytics - Firebase in React Native?
How do I disable server mode and send it via device mode?


<img width="747" alt="Screen Shot 2020-12-09 at 8 24 34 PM" src="https://user-images.githubusercontent.com/13100108/101645667-a1bf4080-3a5c-11eb-85d8-598e66b05f19.png">

**Comment by jariwalabhavesh** [comment]
@JijoBose : Just need to add destination base segment packages.
for example if you want to segment to send event to firebase analytics in device mode. you need to integrate `@segment/analytics-react-native-firebase`

ref:@segment/analytics-react-native-firebase

**Comment by JijoBose** [comment]
> @JijoBose : Just need to add destination base segment packages.
> for example if you want to segment to send event to firebase analytics in device mode. you need to integrate `@segment/analytics-react-native-firebase`
> 
> ref: https://github.com/segmentio/analytics-react-native#packaging-device-mode-destination-sdks

Hi, @jariwalabhavesh , I'm using @segment/analytics-react-native-firebase but still it's the same

```
import analytics from '@segment/analytics-react-native'
import firebase from '@segment/analytics-react-native-firebase'

class SegmentAnalytics {
  constructor() {
    analytics
      .setup(Config.SEGMENT_ID, {
        using: [firebase],
        recordScreenViews: true,
        trackAppLifecycleEvents: true,
        trackAttributionData: true,

        android: {
          flushInterval: 60,
          collectDeviceId: true
        },
        ios: {
          trackAdvertising: true,
          trackDeepLinks: true
        }
      })
      .then(() => console.log('Analytics is ready'))
      .catch(err => console.error('Something went wrong', err))
  }

  screen(screenName, options) {
    analytics.screen(screenName, options)
  }
}

const Segment = new SegmentAnalytics()
export default Segment
```

**Comment by jariwalabhavesh** [comment]
@JijoBose : Have enable firebase destination?
refer this comment:
https://github.com/segmentio/analytics-react-native/issues/9#issuecomment-707747466

**Comment by JijoBose** [comment]
> @JijoBose : Have enable firebase destination?
> refer this comment:
> [#9 (comment)](https://github.com/segmentio/analytics-react-native/issues/9#issuecomment-707747466)

Yes, It's already enabled.
also noticed that screen tracking is sending as 'cloud' mode instead of 'device' mode
